src/build/smart-contracts.md

```diff
@@ -0,0 +1,484 @@
+
+### Functions present in CW20 base standard
+
+#### Actions
+
+- **mint** - Create new tokens, update total supply, and add them to the balance of recipient, as long as it does not exceed the cap. The caller must be a minter.
+
+- **transfer** - Moves the specified amount of tokens from the sender account to the recipient account. This is designed to send to an address controlled by a private key and does not trigger any actions on the recipient if it is a contract.
+
+- **burn** - Remove the specified amount tokens from the balance of the sender.
+
+- **increaseAllowance** - Set or increase the allowance limit of spender for a given caller such that spender may access up to amount + current allowance tokens from the sender account.
+
+- **decreaseAllowance** - Decrease or clear the allowance limit of spender for a given caller such that spender may access up to the current allowance minus the amount tokens from the sender account.
+
+- **transferFrom** - Used to transfer token from src to target and caller must be the address who is pre approved by the src. Pre approve of address is done by increaseAllowance. This makes use of an allowance and if there was a valid, un-expired pre-approval for the sender, then moves amount tokens from owner to recipient and deduct it from the available allowances.
+
+#### Queries
+
+- **balance** - Returns the balance of the given address.
+
+- **allowance** - Returns the available allowance that spender can access from the owner's account.
+
+- **allAllowances** - Returns a list of all address who are allowed to spend(non-expired allowances) by given owner address.
+
+- **allAccounts** - Returns paginated list of all accounts with their balance equals to 0.
+
+- **tokenInfo**- Returns the name and symbol of a token.
+
+- **minter** - Returns a the address of a minter for a given contract(who and how much can be minted).
+
+### Download Cudos node and code
+
+- Setup a CUDOS\_NODED using binary builder mentioned here [https://github.com/CudoVentures/cudos-network-docs/blob/Table-of-Content-folders/src/docs/build-and-earn/testnet-guides/start-binaries.md](https://github.com/CudoVentures/cudos-network-docs/blob/Table-of-Content-folders/src/docs/build-and-earn/testnet-guides/start-binaries.md)
+- Download [cosmwasm-plus](https://github.com/CosmWasm/cw-plus) contract and compile the contracts using docker. Please make sure that you run the docker command inside the root folder.
+
+### Compile the contracts
+
+- Go to cosmwasm-plus folder and run the following command:
+```
+foo@bar:~$ docker run --rm -v "$(pwd)":/code \
+--mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target \
+--mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
+cosmwasm/workspace-optimizer:0.11.3
+```
+
+### Deployment and instantiation
+
+1. Set the environment variables:
+```
+foo@bar:~$ RPC="http://localhost:26657"
+foo@bar:~$ CHAIN_ID="localnet"
+foo@bar:~$ export NODE="--node $RPC"
```

Comment:
lots of (~30) comments were lost in this commit (perhaps due to the document structure changes) but this comment is still applicable to the latest updates/modifications as well

Reply:
it makes sense to have all variables either exported or unexported - depending on what you want to achieve

furthermore: for the NODE, TXFLAG and KEYRING: my previous comment was that i'm not sure if this will work if set this way - did anyone successfully tests and if yes: how?

specifically for KEYRING: it's now [re]defined being in two places (i think the second one is correct)

Reply:
we can change RPC to https://sentry1.gcp-uscentral1.cudos.org:26657